src/core/jit/graphql_executor.rs

```diff
@@ -68,10 +68,13 @@ impl JITExecutor {
                 })
             })
             .await;
-        let val = out.unwrap_or_default();
-        Arc::into_inner(val).unwrap_or_else(|| {
-            Response::from_errors(vec![ServerError::new("Deduplication failed", None)])
-        })
+
+        // Using Arc it's easy to fool the dedupe, but it can be problematic if the
+        // underlying value isn't cloneable. To ensure deduplication works
+        // correctly and to maintain consistency, we clone the response before
+        // sending it out. This approach guarantees that response value is
+        // available to all deduplicated requests.
```

Comment:
better not to store data as `Arc` in that case, please, drop it from dedupe `dedupe_operation_handler`

Reply:
create a simple wrapper with clone implementation for `Response` to bypass the type restrictions.

Reply:
done, using `Lift` for cloning.